test/lib/mayaUsd/fileio/testAddMayaReference.py

```diff
@@ -21,130 +21,306 @@
 import mayaUsd
 import mayaUtils
 
-from pxr import Tf
+from pxr import Tf, Usd, Kind
 
 from maya import cmds
 from maya import standalone
-from mayaUsdLibRegisterStrings import getMayaUsdLibString
-from mayaUSDRegisterStrings import getMayaUsdString
+import mayaUsdAddMayaReference
 
 import unittest
 
 class AddMayaReferenceTestCase(unittest.TestCase):
     '''Test Add Maya Reference.
     '''
     pluginsLoaded = False
+    mayaSceneStr = None
+    stage = None
+    kDefaultNamespace = 'simpleSphere'
 
     @classmethod
     def setUpClass(cls):
         fixturesUtils.readOnlySetUpClass(__file__, loadPlugin=False)
         if not cls.pluginsLoaded:
             cls.pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
 
+        # Create a pure Maya scene to reference in.
+        cls.mayaSceneStr = cls.createSimpleMayaScene()
+
     @classmethod
     def tearDownClass(cls):
         standalone.uninitialize()
 
     def setUp(self):
+        # Start each test with a new scene with empty stage.
         cmds.file(new=True, force=True)
+        import mayaUsd_createStageWithNewLayer
+        self.proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        self.stage = mayaUsd.lib.GetPrim(self.proxyShapePathStr).GetStage()
```

Comment:
Added new test functions for each feature to test. The test class will create the temp maya scene (that we reference in). Each test will start with new scene and empty stage with layer.